ngx_rtmp_relay_module.c

```diff
@@ -361,7 +390,21 @@ ngx_rtmp_relay_create_connection(ngx_rtmp_conf_ctx_t *cctx, ngx_str_t* name,
         goto clear;
     }
 
-    if (ngx_rtmp_relay_copy_str(pool, &rctx->url, &target->url.url) != NGX_OK) {
+    url = &target->url;
+
+    if(session && ngx_strlchr(url->url.data, url->url.data + url->url.len, '$')) {
+        ngx_str_t *u = ngx_pcalloc(pool, sizeof(ngx_str_t));
+        if(ngx_rtmp_eval(session, &target->url.url, ngx_rtmp_relay_eval, u, racf->log) == NGX_OK) {
+            url = ngx_pcalloc(pool, sizeof(ngx_url_t));
+            url->default_port = 1935;
+            url->uri_part = 1;
+            url->url = *u;
+            if(ngx_parse_url(pool, url) != NGX_OK) {
+                goto clear;
+            }
+        }
```

Comment:
Better to move these lines to ngx_rtmp_relay_push/ngx_rtmp_relay_pull.

Reply:
OK. But why not move it into ngx_rtmp_relay_create_remote_ctx?

Reply:
If it is in ngx_rtmp_relay_create_remote_ctx we still need a extra parameter url passed to the ngx_rtmp_relay_create_connection as we cannot simply replace the url in the target: the target is fetched from the conf.

Reply:
Yes, that's the very thing I added:

    save = target->url;
    …
    target->url = save;